lib/screens/explore_screens/unplant_seeds/interactor/viewmodels/unplant_seeds_bloc.dart

```diff
@@ -23,7 +26,13 @@ class UnplantSeedsBloc extends Bloc<UnplantSeedsEvent, UnplantSeedsState> {
     } else if (event is OnAmountChange) {
       yield AmountChangerMapper().mapResultToState(state, event.amountChanged);
     } else if (event is OnMaxButtonTap) {
-      yield AmountChangerMapper().mapResultToState(state, event.maxAmount);
+      final double plantedBalance = state.plantedBalance?.amount ?? 0;
```

Comment:
why not do this inside the mapper?

Reply:
cause I use the same mapper for both maxButton and changeValue